waspc/src/Wasp/Analyzer/Parser/Tokenizer.hs

```diff
@@ -0,0 +1,26 @@
+module Wasp.Analyzer.Parser.Tokenizer
+  ( isValidWaspIdentifier,
+  )
+where
+
+import Control.Monad.Except (runExcept)
+import Control.Monad.State (evalStateT)
+import Wasp.Analyzer.Parser.Lexer (alexScanTokens)
+import Wasp.Analyzer.Parser.Monad (Parser, makeInitialState)
+import Wasp.Analyzer.Parser.Token (Token (Token), TokenType (TIdentifier))
+
+isValidWaspIdentifier :: String -> Bool
+isValidWaspIdentifier str = case tokenize str of
+  Just [Token (TIdentifier name) _ _] -> length name == length str
+  _ -> False
+
+tokenize :: String -> Maybe [Token]
+tokenize str = alexScanTokens str >>= extractTokens
+
+extractTokens :: [Parser Token] -> Maybe [Token]
+extractTokens = mapM extractToken
+
+extractToken :: Parser Token -> Maybe Token
+extractToken tokenParser = case runExcept $ evalStateT tokenParser $ makeInitialState "" of
```

Comment:
To my understanding, this `Either` will never really fail because all tokens are valid. Should I just do `fromRight` and ignore it (and avoid `mapM` in `extractTokens`)?

Reply:
It will not fail because you are not throwing any ParserErrors (instead you return Nothing). But due to comments above I think we will change that.

If we didn't do the modification, then yeah, you could do as you described. But I feel it is all a bit wrong and this Maybe is really redundant here, we should use directly Parser, as I described above.